Replace TOML with JSON for config in the command line.

ntoll opened the pull request:
This PR introduces two changes:

* As per our recent discussion, configuration is expressed as JSON not TOML.
* Update to tests to reflect this change.

That's it..!

(I removed "created_on" because we actually end up getting that from the API when the app is registered.)

fpliger commented on src/pyscript/__init__.py:
```diff
@@ -1,19 +1,20 @@
-if not DATA_DIR.exists():
-    DATA_DIR.mkdir(parents=True)
-    CONFIG_FILE.touch(exist_ok=True)
+CONFIG_FILE = DATA_DIR / Path("config.json")
```
why the name change here?

ntoll replied:
This is a configuration file for the `pyscript` command itself, not the config filename for a PyScript project.

fpliger reviewed (approved):
Left a minor comment but lgtm! Thank you!

fpliger commented:
@ntoll hope I did the right thing... change the name of the default config to `pyscript.json` as we discussed but I'm not sure the line I commented in my review is actually doing. So I may be wrong. Please fix if I misinterpreted and feel free to merge... :)

ntoll commented:
@fpliger I've added a new aspect to this PR of which you should be aware since it relates to https://github.com/anaconda/pyscript-api/pull/7 (changes incoming on that in a moment):

* The default filename for PyScript projects is defined in the `pyscript.config["project_config_filename"]` default setting.
* Updated tests to reflect this.
* This setting, can be read by plugins (such as the `pyscript-api` project) to find the expected location of a project's settings.

Rationale... it's the core `pyscript` command's `create` that defines what this project filename is, and so plugins should use this value, rather than define it themselves.

fpliger commented:
Perfect. That's definitely better for standardizing the config filename for the cli AND plugins
